app/controllers/ownerships_controller.rb

```diff
@@ -44,6 +44,58 @@ def destroy
     @ownership.destroy
   end
 
+  # POST /ownerships/batch
+  # {
+  #   "ownership": [
+  #      {
+  #         "model_id": "xxxxxxxxxxx",
+  #         "model_type": "THING",
+  #         "owner_id": "jeff@jeff.com"
+  #      },
+  #      ...
+  #   ]
+  # }
+  def create_batch
+    # Batch upsert does not seem to be possible.
+    # Instead, separate out the creates and updates.
+    #   Batch update does not seem to be possible.
+    #   However, batch create is definitely possible.
+    modelidhash = {}
+    # I don't know what the expected behaviour is if the same model id is passed twice,
+    #  so abort if a duplicate is found.
+    params.require(:ownership).each do |item|
```

Comment:
In Rails normally this parameters processing/validation goes into its own method. This allows us to reuse this method in case we want to add new actions that needs the same parameter processing.  You can call it something like 'batch_parameters' and defined it private.

About the parameter validation, there's possibly a way of doing this validation on the content of the array in the ownership parameters without needing to iterate over them. Take a look at the example in:
https://github.com/rails/strong_parameters#nested-parameters

Unfortunately the treatment of duplicates that you are doing cannot be done using the way is done in the documentation. Maybe you can perform it after the validation step.